handler-proxy/src/main/java/io/netty/handler/proxy/HttpProxyHandler.java

```diff
@@ -127,13 +128,36 @@ protected void removeDecoder(ChannelHandlerContext ctx) throws Exception {
     @Override
     protected Object newInitialMessage(ChannelHandlerContext ctx) throws Exception {
         InetSocketAddress raddr = destinationAddress();
-        final String host = NetUtil.toSocketAddressString(raddr);
+
+        String hostString = PlatformDependent.javaVersion() >= 7 ? raddr.getHostString() : raddr.getHostName();
```

Comment:
@slandelle should we make this a utility method in the http package ? Like in HttpUtils ? I think this is something that is useful in general.

Reply:
Can do, sure. But does it make sense to introduce this now while work on Netty 5 targeting Java 8 is about to start?

Reply:
@slandelle I think it still does as netty 5 is most likely still be 1 year away

Reply:
ok

Reply:
This utility would be better suited for NetUtil than HttpUtil, don't you think?

Reply:
Not sure... isnt it kind of related how you format stuff for the http header ?

Reply:
Ah, ok, I thought you were only talking about the first line above your comment, not the whole block. Will do.